CMakeLists.txt

```diff
@@ -147,19 +146,29 @@ if(NOT CORENRN_ENABLE_TIMEOUT)
   add_definitions("-DDISABLE_TIMEOUT")
 endif()
 
-if ( CORENRN_ENABLE_SONATA AND CORENRN_ENABLE_REPORTINGLIB )
-  message(FATAL_ERROR "Can't have enabled both SONATA reports and reportinglib")
-endif()
-
-if(CORENRN_ENABLE_REPORTINGLIB)
-  find_package(reportinglib REQUIRED)
+if(CORENRN_ENABLE_REPORTING)
+  find_package(reportinglib)
+  find_package(sonata)
+  if(NOT reportinglib_FOUND AND NOT sonata_FOUND)
+    message(FATAL_ERROR "NO reporting library was found")
+  elseif(reportinglib_FOUND AND sonata_FOUND)
+    add_definitions("-DENABLE_REPORTINGLIB")
```

Comment:
Macros should be prefixed by their project. Something like `CORENRN_ENABLE_SONATA_REPORTS`

Reply:
I didn't add CORENRN_ prefix to the macro names because none of the other macros have the prefix, but I like the change REPORTINGLIB for BIN_REPORTS

Reply:
fair! we can do that change separately. By the way there is a separate PR #167 but that needs some extra work. Let's take this separately.